installer/pkg/nextgen/asset.go

```diff
@@ -0,0 +1,6 @@
+package nextgen
+
+type Asset interface {
+	Dependencies() []Asset
+	Generate([]*AssetState) (*AssetState, error)
```

Comment:
It would nice if we Generate could be entire `State` https://github.com/openshift/installer/blob/master/Documentation/design/assetgeneration.md#state

Reply:
What would be nice about getting the entire `State`? The asset has already stated what its dependencies are. If it needs anything from `State` beyond those dependencies, then it is using a dependencies that it did not declare. The `AssetStore` has already fetched all of the states for all of the dependencies, so there is no need for the asset to relocate them from the entire `State`.